docs/ruleset_functions/compare_swh_dist_systems_and_components.md

```diff
@@ -0,0 +1,54 @@
+
+Function Call:
+
+- **compare_context_pair** - there is no RDS for this function, but it is a function developed for Rule 1-6 that compares two elements  
+- **get_component_by_id**
+
+Data Lookup: None
+
+Logic:
+  - create the list errors: `errors = []`
+  - use the function get_swh_components_associated_with_each_swh_distribution_system to get a dictionary of SWH equipment associated with each Distribution System for model 1 (pumps, tanks, SWH use, distribution system, etc): `RMD1_swh_system_and_equip_dict = get_swh_components_associated_with_each_swh_distribution_system(RMD1)`
+  - use the function get_swh_components_associated_with_each_swh_distribution_system to get a dictionary of SWH equipment associated with each Distribution System for model 1 (pumps, tanks, SWH use, distribution system, etc): `RMD2_swh_system_and_equip_dict = get_swh_components_associated_with_each_swh_distribution_system(RMD2)`
+
+  - check if this distribution ID is in model 1: `if RMD1_swh_system_and_equip_dict[swh_distribution_id]:`
+    - get the dictionary of all SWH equipment (pumps, tanks, SWH use, dist system, etc) for the given swh_distribution_id for RMD1: `RMD1_swh_equipment_dict = RMD1_swh_system_and_equip_dict[swh_distribution_id]`
+    - get the actual distribution equipment associated with this ID from model 1: `RMD1_swh_distribution = get_component_by_id(RMD1, swh_distribution_id)`
+  - otherwise, return FALSE by adding an error to the list: `else: errors << "SWH Distribution System " + swh_distribution_id + " not found in one of the two RMDs"`
+
+  - check if this distribution ID is in model 2: `if RMD2_swh_system_and_equip_dict[swh_distribution_id]:`
+    - get the dictionary of all SWH equipment (pumps, tanks, SWH use, dist system, etc) for the given swh_distribution_id for RMD2: `RMD2_swh_equipment_dict = RMD2_swh_system_and_equip_dict[swh_distribution_id]`
+    - get the actual distribution equipment associated with this ID from model 1: `RMD2_swh_distribution = get_component_by_id(RMD2, swh_distribution_id)`
+  - otherwise, return FALSE by adding an error to the list: `else: errors << "SWH Distribution System " + swh_distribution_id + " not found in one of the two RMDs"`
+
+
+  - Compare the distribution in the two models using the function compare_context_pair.  compare_context_pair is recursive, so by sending the function the distribution systems, it is also checking the tanks and piping that are child objects of the distribution systems.  If the comparison fails, we add an error to the errors list indicating the id of the swh distribution system that failed comparison: `if !compare_context_pair(RMD1_swh_distribution, RMD2_swh_distribution, $, extra_schema_for_SWH_comparison.json, true, compare_context_str, errors): errors << "Comparison of SWH Distribution System: " + swh_distribution_id + " failed."`
```

Comment:
`$`? I suggest ignoring any attempt for proper syntax to call compare_context_pair because there is no RDS for it - I would just describe the arguments that you know you want to pass to it

Reply:
any suggestions on how to communicate this effectively?